api_tests/nodes/views/test_node_addons.py

```diff
@@ -0,0 +1,714 @@
+    def set_folder_url(self):
+        self.folder_url = '/{}nodes/{}/addons/{}/folders/'.format(
+            API_BASE, self.node._id, self.short_name
+        )
+
+    def test_folder_list_GET_expected_behavior(self):
+        wrong_type = self.short_name != 'googledrive'
+        res = self.app.get(
+            self.folder_url,
+            auth=self.user.auth,
+            expect_errors=wrong_type)
+
+        if not wrong_type:
+            addon_data = res.json['data'][0]['attributes']
+            assert_equal(addon_data['path'], '/')
+            assert_equal(addon_data['kind'], 'folder')
+            assert_equal(addon_data['name'], '/ (Full Google Drive)')
+            assert_equal(addon_data['folder_id'], 'FAKEROOTID')
+        if wrong_type:
+            assert_in(res.status_code, [404, 405])
+
+
+    def test_folder_list_raises_error_if_PUT(self):
+        res = self.app.put_json_api(self.folder_url, {
+            'id': self.short_name,
+            'type': 'node-addon-folders'
+            }, auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_folder_list_raises_error_if_PATCH(self):
+        res = self.app.patch_json_api(self.folder_url, {
+            'id': self.short_name,
+            'type': 'node-addon-folders'
+            }, auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_folder_list_raises_error_if_DELETE(self):
+        res = self.app.delete(
+            self.folder_url,
+            auth=self.user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 405)
+
+    def test_folder_list_GET_raises_error_noncontrib_not_public(self):
+        noncontrib = AuthUserFactory()
+        res = self.app.get(
+            self.folder_url,
+            auth=noncontrib.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_folder_list_GET_raises_error_writecontrib_not_authorizer(self):
+        write_user = AuthUserFactory()
+        self.node.add_contributor(write_user, permissions=[WRITE], auth=self.auth)
+        res = self.app.get(self.folder_url, 
+            auth=write_user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_folder_list_GET_raises_error_admin_not_authorizer(self):
+        admin_user = AuthUserFactory()
+        self.node.add_contributor(admin_user, permissions=[ADMIN], auth=self.auth)
+        res = self.app.get(self.folder_url, 
+            auth=admin_user.auth,
+            expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+
+class NodeAddonTestSuiteMixin(NodeAddonListMixin, NodeAddonDetailMixin, NodeAddonFolderMixin):
+    def set_urls(self):
+        self.set_setting_list_url()
+        self.set_setting_detail_url()
+        self.set_folder_url()
+
+    def should_expect_errors(self, success_types=['CONFIGURABLE', 'OAUTH']):
```

Comment:
Even for a simple use case such as this, it's a good idea to avoid mutable default argument values. Eg use a tuple instead of a list for `success_types`.